accord-core/src/main/java/accord/local/cfk/CommandsForKey.java

```diff
@@ -1604,18 +1639,21 @@ public CommandsForKey updateUniqueHlc(long minUniqueHlc)
         if (maxUniqueHlc <= bounds.gcBefore.hlc() && bounds.gcBefore.is(HLC_BOUND))
             return this;
 
-        return new CommandsForKey(key, bounds, byId, minUndecidedById, maxAppliedPreBootstrapWriteById, committedByExecuteAt, maxAppliedWriteByExecuteAt, minUniqueHlc, loadingPruned, prunedBeforeById, unmanageds);
+        return new CommandsForKey(key, bounds, byId, minUndecidedById, maxAppliedPreBootstrapWriteById, committedByExecuteAt, maxAppliedWriteByExecuteAt, minUniqueHlc, loadingPruned, prunedBeforeById, unmanageds, true);
     }
 
-    public CommandsForKey setDurable(TxnId txnId)
+    public CommandsForKey setDurable(TxnId txnId, Durability durability)
     {
         TxnInfo txn = get(txnId);
-        if (txn == null || !txn.is(APPLIED_NOT_DURABLE))
+        if (txn == null || !durability.isDurablyCommitted())
```

Comment:
I might be misunderstanding: do we _not_ want to set durable in CFK if not yet durably committed? Should we even be calling this method then?

Reply:
I don't quite follow, but if you're asking whether this method should not have been invoked, the answer is we already should not invoke it if we think it won't help, but the method signature doesn't specify what Durability it would be valid to invoke it with, and we don't want to corrupt the state if we for some reason invoke setDurability with a Durability that doesn't imply anything useful for CommandsForKey, so the most sensible thing IMO is to simply do nothing if the parameter doesn't tell us anything we want to know.